src/main/scala/util/package.scala

```diff
@@ -33,6 +33,29 @@ package object util {
     }
 
     def asUInt(): UInt = Cat(x.map(_.asUInt).reverse)
+
+    def rotate(n: Int): Seq[T] = x.drop(n) ++ x.take(n)
+
+    def rotate(n: UInt): Seq[T] = {
+      require(isPow2(x.size))
+      val amt = n.padTo(log2Ceil(x.size))
+      (x /: (0 until log2Ceil(x.size)))((r, i) => (r.rotate(1 << i) zip r).map { case (s, a) => Mux(amt(i), s, a) })
+    }
+  }
+
+  // allow bitwise ops on Seq[Bool] just like UInt
+  implicit class SeqBoolBitwiseOps(val x: Seq[Bool]) extends AnyVal {
```

Comment:
@mwachs5 I feel like you were just asking for this. You should get automagically if you've imported `rocketchip.util`